templates/data-store-pv.yaml

```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 Crown Copyright
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Keep track of the root . inside control structures
+{{- $root := . }}
+# Create a number of persistent volumes for each configured data-store
+{{- range $store := .Values.global.persistence.dataStores }}
+# New PV for each dataStores entry
+---
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  # This range item is referenced $store and the root item referenced $root
+  name: {{ $store.name }}
+  {{- if eq $root.Values.hosting "aws" }}
+  labels:
+    type: amazonEBS
+  {{- end }}
+spec:
+  accessModes:
+  - ReadOnlyMany
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: {{ $store.name }}
+  capacity:
+    storage: {{ $store.storage }}
+  {{- if eq $root.Values.hosting "local" }}
+  hostPath:
+    # Directory location on host
+    path: {{ $store.hostPath }}
+    type: DirectoryOrCreate
```

Comment:
Is this definitely required. Am sure in another PR @dev-958 removed this field in a different PV as it wasn't required

Reply:
DirectoryOrCreate is required over using Directory on my build of WSL